shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/watchers/DeploymentEvent.java

```diff
@@ -0,0 +1,50 @@
+package com.redhat.service.bridge.shard.operator.watchers;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.Watcher;
+import io.javaoperatorsdk.operator.processing.event.DefaultEvent;
+
+public class DeploymentEvent extends DefaultEvent {
+
+    private final Watcher.Action action;
+    private final Deployment deployment;
+
+    public DeploymentEvent(Watcher.Action action, Deployment resource, DeploymentEventSource deploymentEventSource) {
+        // TODO: this mapping is really critical and should be made more explicit by the java operator sdk
+        super(resource.getMetadata().getOwnerReferences().get(0).getUid(), deploymentEventSource);
```

Comment:
Do we have an open issue for them?

Reply:
Hey @ricardozanini , tbh I took it from the `samples` provided by the sdk https://github.com/java-operator-sdk/samples/blob/ad32eb0cbf525b9affd11a44d7f3c6c24ba5c719/tomcat/src/main/java/io/javaoperatorsdk/operator/sample/DeploymentEvent.java#L14 

My understanding was that this is something that can be improved on api level of the sdk, or maybe it was for the samples themselves? 
But actually it is really critical, the `ownerReference` mapping is mandatory otherwise our controller does not get called when the resource is updated

Reply:
Got it. I was just wondering if it's worth talking to them about that. I agree with you, it's pretty critical.